smithy-model/src/test/java/software/amazon/smithy/model/shapes/ModelSerializerTest.java

```diff
@@ -52,15 +52,18 @@ public class ModelSerializerTest {
     public Stream<DynamicTest> generateTests() throws IOException, URISyntaxException {
         return Files.list(Paths.get(
                         SmithyIdlModelSerializer.class.getResource("ast-serialization/cases").toURI()))
-                .filter(path -> !path.toString().endsWith(".1.0.json"))
+                .filter(path -> !path.toString().matches("^.*\\.\\d\\.0\\.json$"))
                 .map(path -> DynamicTest.dynamicTest(path.getFileName().toString(), () -> testRoundTrip(path)));
     }
 
     public void testRoundTrip(Path path) {
         Model model = Model.assembler().addImport(path).assemble().unwrap();
         ModelSerializer serializer = ModelSerializer.builder().build();
         ObjectNode actual = serializer.serialize(model);
-        ObjectNode expected = Node.parse(IoUtils.readUtf8File(path)).expectObjectNode();
+        // Use explicit 2.0 comparison file if present; otherwise, default to original path
+        Path potentialExpectedFile2 = Paths.get(path.toString().replace(".json", ".2.0.json"));
+        Path expectedFile2 = Files.exists(potentialExpectedFile2) ? potentialExpectedFile2 : path;
```

Comment:
i'm a bit unsure about all this logic and the file names. Seems like there should be a better way to do this.

Reply:
Will be refactored into more verbose / separate test factories.